src/NServiceBus.AcceptanceTests/Sagas/When_using_ReplyToOriginator.cs

```diff
@@ -10,57 +10,83 @@
+                            SomeId = someId
+                        }))
+                )
                 .Done(c => c.Done)
                 .Run();
 
             Assert.AreEqual(MessageIntentEnum.Reply, context.Intent);
+            Assert.AreEqual(messageId, context.ReceivedCorrelationId, "Message id should be preserved and used as correlation id on reply so that things like callbacks work properly");
         }
 
         public class Context : ScenarioContext
         {
             public MessageIntentEnum Intent { get; set; }
+            public string ReceivedCorrelationId { get; set; }
             public bool Done { get; set; }
+            public bool GotFirstMessage { get; set; }
         }
 
         public class Endpoint : EndpointConfigurationBuilder
         {
             public Endpoint()
             {
-                EndpointSetup<DefaultServer>(config => config.EnableFeature<TimeoutManager>());
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.EnableFeature<TimeoutManager>();
+                    config.LimitMessageProcessingConcurrencyTo(1); //to avoid race conditions with the start and second message
+                });
             }
 
             public class RequestingSaga : Saga<RequestingSaga.RequestingSagaData>,
                 IAmStartedByMessages<InitiateRequestingSaga>,
-                IHandleMessages<AnotherRequest>
+                IHandleMessages<MessageThatWillCauseSagaToReplyToOriginator>
             {
+                public Context TestContext { get; set; }
+
                 public Task Handle(InitiateRequestingSaga message, IMessageHandlerContext context)
                 {
-                    Data.CorrIdForResponse = message.SomeCorrelationId; //wont be needed in the future
+                    Data.CorrIdForResponse = message.SomeId; //wont be needed in the future
+                    TestContext.GotFirstMessage = true;
 
-                    return context.SendLocal(new AnotherRequest
-                    {
-                        SomeCorrelationId = Data.CorrIdForResponse //wont be needed in the future
-                    });
+                    return Task.FromResult(0);
                 }
 
-                public async Task Handle(AnotherRequest message, IMessageHandlerContext context)
+                public async Task Handle(MessageThatWillCauseSagaToReplyToOriginator message, IMessageHandlerContext context)
                 {
                     await ReplyToOriginator(context, new MyReplyToOriginator());
                     MarkAsComplete();
                 }
 
                 protected override void ConfigureHowToFindSaga(SagaPropertyMapper<RequestingSagaData> mapper)
                 {
-                    mapper.ConfigureMapping<InitiateRequestingSaga>(m => m.SomeCorrelationId)
+                    mapper.ConfigureMapping<InitiateRequestingSaga>(m => m.SomeId)
                         .ToSaga(s => s.CorrIdForResponse);
```

Comment:
that naming seems weird? Shouldn't that be just `SagaCorrelationId`? It has nothing to do with the response itself?